de.tu_bs.cs.isf.e4cf.parts.project_explorer/src/de/tu_bs/cs/isf/e4cf/parts/project_explorer/wizards/NewFileDialog.java

```diff
@@ -0,0 +1,83 @@
+package de.tu_bs.cs.isf.e4cf.parts.project_explorer.wizards;
+
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import de.tu_bs.cs.isf.e4cf.core.stringtable.E4CFileTable;
+import de.tu_bs.cs.isf.e4cf.core.util.services.RCPImageService;
+import de.tu_bs.cs.isf.e4cf.parts.project_explorer.stringtable.FileTable;
+import javafx.scene.control.Button;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.TextField;
+import javafx.scene.control.TextInputDialog;
+import javafx.stage.Stage;
+
+/** A wizard that lets the user create a new file. */
+public class NewFileDialog {
+
+	private Path parentPath;
+	private Path targetPath;
+	private TextInputDialog dialog;
+
+	// the regex to check if filename is valid.
+	private static final String FILE_NAME_REGEX = "[a-zA-Z0-9-_#]*[.]{1}[a-zA-Z0-9]+";
+
+	/**
+	 * @param parentPath
+	 * @param imageService
+	 */
+	public NewFileDialog(Path parentPath, RCPImageService imageService) {
+		this.parentPath = parentPath;
+		dialog = new TextInputDialog("New File");
+
+		final Stage stage = (Stage) dialog.getDialogPane().getScene().getWindow();
+		stage.getIcons().add(imageService.getFXImage(null, E4CFileTable.FRAMEWORK_LOGO_SMALL).getImage());
+
+		dialog.setTitle("Create New File...");
+		dialog.setHeaderText("Root Directory: " + parentPath.getFileName() + "\nEnter the new file name:");
+		dialog.setGraphic(imageService.getFXImage(null, FileTable.NEWFILE_PNG));
+
+		final Button btnOk = (Button) dialog.getDialogPane().lookupButton(ButtonType.OK);
+		btnOk.setDisable(true);
+
+		final TextField input = dialog.getEditor();
+		input.setOnKeyReleased(name -> {
+			if (isFilenameValid(input.getText())) {
+				btnOk.setDisable(false);
+			} else {
+				btnOk.setDisable(true);

```

Comment:
The user should be notified why the file cannot be created.
E.g. in an additional label with red font color

Reply:
the dialog is now reworked to be a custom dialog that supports label for displaying the information